### DISCUSSION TITLE
LORIS V26 upgrade issue

### DISCUSSION THREAD

**Discussion by Johan-Peralta** [aces/Loris#9308] [General]
Hi, I followed the steps below to upgrade a LORIS instance from v25.0.x to v26.0.0:
1. Update to the v26.0.0 version of the code.
2. Run the SQL update patch SQL/Release_patches/25.0_To_26.0_upgrade.sql.
3. Run the command make from your LORIS root directory.

The **make** step yielded this error: 
```
**ERROR in ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/chunks/index.tsx 8:17-56
Module not found: Error: Can't resolve '../protocol-buffers/chunk_pb' in '/var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/chunks'

webpack compiled with 1 error
make: [Makefile:6: all] Error 1**
```

The **loris-error.log** captures this other error:
`**[Tue Jul 16 20:52:44.235200 2024] [php:warn] [pid 1484] [client 123.12.456.34:38647] PHP Warning:  Undefined array key 1 in /var/www/loris/modules/statistics/php/charts.class.inc on line 129**`

Any ideas of what could be the root cause?

Thanks in advance for any help with this issue,
Johan

**Comment by ridz1208** [comment]
Tagging @laemtl and @jeffersoncasimir as they might know more about this. As far as I know this is because the protocol buffer is not installed. This module readme seems to suggest to follow github installation instructions 
https://github.com/aces/Loris/tree/main/modules/electrophysiology_browser#installation-requirements-to-use-the-visualization-features

However protocol buffer should not be required unless EEG browser s being set up according to https://github.com/aces/Loris/pull/7676 
@laemtl might have more details here

**Comment by driusan** [comment]
Until you've successfully run make I wouldn't make much of any other warnings, because if make hasn't run and there's a mismatch between the version of the code running on the backend and on the frontend anything can happen, but theoretically I think this PR should fix the warning from statistics:

https://github.com/aces/Loris/pull/9310/files

**Comment by Johan-Peralta** [comment]
Gentlemen, 
1. I am not sure how the EEG browser setup got involved in this upgrade, but after installing the protocol buffer the **make** completed with a few warnings and not errors.

2. After making the change to charts.class.inc suggested in -> https://github.com/aces/Loris/pull/9310/files. 

3. The loris-error.log captures is now registering this error:
`[Wed Jul 17 18:13:58.009053 2024] [php:notice] [pid 2773] [client 123.12.456.34:21765] LORIS Deprecation Warning: The getAllInstruments() function of the Utility class has been deprecated. This function does not load the most accurate Full_name value and should be replaced by getInstrumentNamesList() of the NDB_BVL_Instrument class., referer: http://AA.BBB.CC.DD/statistics/statistics_site/undefined/statistics`

4. LORIS hangs after the user login:
<img width="892" alt="image" src="https://github.com/user-attachments/assets/8358dead-0b2b-48fd-99bb-f814e0ed42f7">
 
Please advise on how to deal with the getAllInstruments() function deprecation.

Thanks in advance for any help with this issue,
Johan

**Comment by Johan-Peralta** [comment]
Update on this V26 upgrade issues:

a. Despite the multiple warnings on the getAllInstruments() function deprecation being registered/capture on the loris-error.log, LORIS V26 is working when accessed with the Firefox and Chrome browsers.

b. The LORIS V26 hanging, after the login in, only occurs when accessing LORIS with the latest MS Edge browser.

c. When invoking the EEG Browser module, these messages are being capture on the loris-error.log:
```
[Fri Jul 19 13:05:51.450019 2024] [php:notice] [pid 21962] [client] Instrument picture_game_debrief does not seem to be a valid instrument.
[Fri Jul 19 13:05:51.450076 2024] [php:notice] [pid 21962] [client] Instrument picture_game_training does not seem to be a valid instrument.
[Fri Jul 19 13:05:51.450113 2024] [php:notice] [pid 21962] [client] Instrument ravens does not seem to be a valid instrument.
[Fri Jul 19 13:05:51.450156 2024] [php:notice] [pid 21962] [client] Instrument task_information does not seem to be a valid instrument.
```
Any advice on how to set the instruments referred in the above messages will be greatly appreciated.

Thanks in advance,
Johan

**Comment by Johan-Peralta** [comment]
Update on this upgrade to V26:

The LORIS V26 hanging was due to caching, clearing the MS Edge browser cache resolved the issue.

Despite the warnings, LORIS V26 is working fine in the MS Edge, Firefox and Chrome browsers.

Thanks for the help.

**Comment by ridz1208** [comment]
Hey @Johan-Peralta 

1. Although that function throws a deprecation warning it is still technically functional, the warning is just a flag that the function will be removed in the next LORIs iteration (probably replaced by another function that already exists) you can ignore these warnings
2. Happy to hear LORIs works on MS edge but just to be clear we officially support Firefox and Chrome. We don't generally go out of our way to support edge if something doesnt work on it.
3. As for instruments. I would recommend looking at the documents here https://github.com/aces/Loris/tree/main/docs/wiki/01_STUDY_PARAMETERS_SETUP/02_Clinical_Instruments and some of the examples in https://github.com/aces/Loris/tree/main/raisinbread/instruments

**Comment by Johan-Peralta** [comment]
[like]  Johan Peralta reacted to your message:
________________________________

**Reply by Johan-Peralta** [reply]
Hi @ridz1208 

1. Thanks for the clarification on the deprecation warning.

2. We know only Firefox and Chrome are the LORIS officially supported browsers. MS Edge is the default browser within our organization, and I do not want to give our Windows users a bad first impression of LORIS.

3. On the instruments, those I referred (picture_game_debrief, picture_game_training, ravens and task_information) appear to be baked into the EEG Browser module. If their definition can be shared, I will install them on our instance (the LORIS EEG Browser has spark interest within our Team).

4. I found the setting **Enable the EEG Browser components** set to **Yes** in the **Configuration -> GUI**, which I think is the reason why the EEG Browser setup was triggered during this upgrade. 

As per usual, thanks for your help and support,
Johan
